src/common.js

```diff
@@ -191,3 +191,8 @@ exports.unexpectedObjToString = function(obj) {
     return baseToString;
   }
 };
+
+exports.isValidJSIdentifier = function(str) {
```

Comment:
Hmmm, this is going to be pretty expensive to pull in the ES5 grammar and parse the string every time we want to check.

Now that I think about it more, I think it might be best to move this function into `toArgumentNameList`, and name it 'isRestrictedJSIdentifier' -- and then only allow ASCII characters (and just check with a simple regex). Otherwise there is a risk that we generate an identifier that the programmer doesn't actually know how to type!